cdk-infra/utils/buckets.ts

```diff
@@ -10,3 +16,149 @@ export const docsBucketNames = [
   'docs-java',
   'docs-node',
 ] as const;
+
+interface CustomBucketProps {
+  scope: Construct;
+  featureName: string;
+  env: AutoBuilderEnv;
+  bucketName: string;
+}
+export function createCustomBucket({ scope, featureName, env, bucketName }: CustomBucketProps): Bucket {
+  const stackBucketName = `${featureName}-${bucketName}-${env}`.toLowerCase();
+  const websiteRoutingRules = bucketName === 'docs-mongodb-org' ? getMongoDocsBucketRoutingRules(env) : undefined;
+
+  const bucket = new Bucket(scope, stackBucketName, {
+    removalPolicy: RemovalPolicy.DESTROY,
+    websiteRoutingRules,
+    bucketName: stackBucketName,
+    websiteIndexDocument: 'index.html',
+    websiteErrorDocument: 'docs-qa/404/index.html',
+    blockPublicAccess: new BlockPublicAccess({
+      blockPublicAcls: false,
+      blockPublicPolicy: false,
+      ignorePublicAcls: false,
+      restrictPublicBuckets: false,
+    }),
+  });
+
+  return bucket;
+}
+
+export function getMongoDocsBucketRoutingRules(env: AutoBuilderEnv): RoutingRule[] {
```

Comment:
I realize this moved from the other file -- I'm just slightly confused about how our buckets end up actually working since the bucket that has all of these defined routing rules is docs-mongodb-org-dotcomprd and I don't see where we append the `-dotcomprd` to the end of the bucket… (To be fair I don't get how that worked in the previous code approach either…)

Reply:
I believe when checking the bucket configurations that all `docs-mongodb-org` buckets have routing rules defined for them. Essentially, the code here is copy-pasted from [here](https://github.com/mongodb/docs-worker-pool/blob/cd75999ac346b96bb8d96a74125fa83d9b85eb90/infrastructure/ecs-main/buckets.yml#L14).

This method doesn't create the bucket, but simply creates the routing rules object that we want to provide for the `docs-mongodb-org` bucket. We append the `env` name for the bucket [here](https://github.com/mongodb/docs-worker-pool/blob/cd75999ac346b96bb8d96a74125fa83d9b85eb90/cdk-infra/lib/constructs/worker/buckets-construct.ts#L24). 

For context, the `getEnv` method is tied to the Snooty/Autobuilder env values such as `dotcomstg`, `dotcomprd`, `prd`, etc.

Reply:
AAAAH. Okay thanks. I was confuzzled! Perfect.